internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -558,6 +571,69 @@ func (r *BMACReconciler) reconcileAgentLabels(bmh *bmh_v1alpha1.BareMetalHost, a
 	return ret, nil
 }
 
+func (r *BMACReconciler) reconcileAgentFencingCredentials(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) (bool, error) {
+	if val, ok := bmh.ObjectMeta.Annotations[BMH_AGENT_FENCING_ANNOTATION]; ok {
+		if agent.Spec.FencingCredentialsSecretRef != val {
+			agent.Spec.FencingCredentialsSecretRef = val
+			return true, nil
+		}
+	}
```

Comment:
I think adding a return here will make the rest of this easier to understand.

```suggestion
	if val, ok := bmh.ObjectMeta.Annotations[BMH_AGENT_FENCING_ANNOTATION]; ok {
		if agent.Spec.FencingCredentialsSecretRef != val {
			agent.Spec.FencingCredentialsSecretRef = val
			return true, nil
		}
		return false, nil
	}
```